Flaky test TestUnweightedDoubleReservoirSample

TestUnweightedDoubleReservoirSample fails sometimes:

```
[ERROR] Tests run: 5921, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1,502.223 s <<< FAILURE! - in TestSuite
[ERROR] testMany(com.facebook.presto.operator.aggregation.reservoirsample.TestUnweightedDoubleReservoirSample)  Time elapsed: 0.069 s  <<< FAILURE!
java.lang.AssertionError: Expected quantized[i] > got: i=0, quantized=35.0, got=35 expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:513)
	at org.testng.Assert.assertTrue(Assert.java:42)
	at com.facebook.presto.operator.aggregation.reservoirsample.TestUnweightedDoubleReservoirSample.testMany(TestUnweightedDoubleReservoirSample.java:71)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

CC: @atavory

--- a/presto-main/src/test/java/com/facebook/presto/operator/aggregation/reservoirsample/TestUnweightedDoubleReservoirSample.java
+++ b/presto-main/src/test/java/com/facebook/presto/operator/aggregation/reservoirsample/TestUnweightedDoubleReservoirSample.java
@@ -58,17 +58,17 @@ public class TestUnweightedDoubleReservoirSample
 
         long streamLength = 1_000_000;
         for (int i = 0; i < streamLength; ++i) {
-            double value = i * 10 + i % 2;
-            reservoir.add(value);
+            reservoir.add(i);
         }
 
         double[] quantized = new double[4];
         for (double sample : reservoir.getSamples()) {
-            int index = (int) (4.0 * sample / (10.0 * streamLength + 1));
+            int index = (int) (4.0 * sample / (streamLength + 1));
             ++quantized[index];
         }
+        int expectedMin = 25;
         for (int i = 0; i < 4; ++i) {
-            assertTrue(quantized[i] > 35, format("Expected quantized[i] > got: i=%s, quantized=%s, got=%s", i, quantized[i], 35));
+            assertTrue(quantized[i] > expectedMin, format("Expected quantized[i] > got: i=%s, quantized=%s, got=%s", i, quantized[i], expectedMin));
         }
     }
 }

--- a/presto-main/src/test/java/com/facebook/presto/operator/aggregation/reservoirsample/TestWeightedDoubleReservoirSample.java
+++ b/presto-main/src/test/java/com/facebook/presto/operator/aggregation/reservoirsample/TestWeightedDoubleReservoirSample.java
@@ -58,17 +58,20 @@ public class TestWeightedDoubleReservoirSample
 
         long streamLength = 1_000_000;
         for (int i = 0; i < streamLength; ++i) {
-            double value = i * 10 + i % 2;
-            reservoir.add(value, 1.0);
+            reservoir.add(i, 1.0);
         }
 
         double[] quantized = new double[4];
+        int count = 0;
         for (double sample : reservoir.getSamples()) {
-            int index = (int) (4.0 * sample / (10.0 * streamLength + 1));
+            ++count;
+            int index = (int) (4.0 * sample / (streamLength + 1));
             ++quantized[index];
         }
+        assertEquals(count, 200, format("Number of samples should be full: got=%s, expected=%s", count, 200));
+        int expectedMin = 25;
         for (int i = 0; i < 4; ++i) {
-            assertTrue(quantized[i] > 35, format("Expected quantized[i] > got: i=%s, quantized=%s, got=%s", i, quantized[i], 35));
+            assertTrue(quantized[i] > expectedMin, format("Expected quantized[i] > got: i=%s, quantized=%s, got=%s", i, quantized[i], 35));
         }
     }
 
@@ -82,17 +85,20 @@ public class TestWeightedDoubleReservoirSample
             reservoir.add(3, 0.00000001);
         }
         for (int i = 0; i < streamLength; ++i) {
-            double value = i * 10 + i % 2;
-            reservoir.add(value, 9999999999.0);
+            reservoir.add(i, 9999999999.0);
         }
 
         double[] quantized = new double[4];
+        int count = 0;
         for (double sample : reservoir.getSamples()) {
-            int index = (int) (4.0 * sample / (10.0 * streamLength + 1));
+            ++count;
+            int index = (int) (4.0 * sample / (streamLength + 1));
             ++quantized[index];
         }
+        assertEquals(count, 200, format("Number of samples should be full: got=%s, expected=%s", count, 200));
+        int expectedMin = 25;
         for (int i = 0; i < 4; ++i) {
-            assertTrue(quantized[i] > 35, format("Expected quantized[i] > got: i=%s, quantized=%s, got=%s", i, quantized[i], 35));
+            assertTrue(quantized[i] > expectedMin, format("Expected quantized[i] > got: i=%s, quantized=%s, got=%s", i, quantized[i], expectedMin));
         }
     }
 }

